ARCHITECTURE.md

```diff
@@ -1,7 +1,18 @@
 Architecture of the library
 ===
 
-    ELF -> Specifications -> Objects -> Links
+```mermaid
```

Comment:
Check out the rich diff!

Reply:
Nice! Shouldn't the arrows be inverted? e.g. ELF -> ProgramSpec. That conveys the flow more naturally and reflects better what was written here before.

Reply:
To me dependency order makes more sense, since that's also how the code ends up being structured. I kept ELF at the left since that makes it easier to explain however!